sacred/randomness.py

```diff
@@ -12,22 +13,29 @@
 def get_seed(rnd=None):
     if rnd is None:
         return random.randint(*SEEDRANGE)
-    return rnd.randint(*SEEDRANGE)
+    elif type(rnd) == opt.np.random.Generator:
+        return rnd.integers(*SEEDRANGE, dtype=int)
+    else:
+        return rnd.randint(*SEEDRANGE)
 
 
 def create_rnd(seed):
     assert isinstance(seed, int), "Seed has to be integer but was {} {}".format(
         repr(seed), type(seed)
     )
     if opt.has_numpy:
-        return opt.np.random.RandomState(seed)
+        if version.parse(opt.np.__version__) >= version.parse('1.19'):
+            return opt.np.random.default_rng(seed)
+        else:
+            return opt.np.random.RandomState(seed)
     else:
         return random.Random(seed)
 
 
 def set_global_seed(seed):
     random.seed(seed)
-    if opt.has_numpy:
+    if opt.has_numpy and \
+            version.parse(opt.np.__version__) >= version.parse('1.19'):
```

Comment:
Why did you add this check? The seed should certainly be set for all versions of sacred, not only for the new one.

Reply:
Sry, wrong direction. np.random.seed is [depreciated / legacy](https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html) in 1.19.